test/test_dataloader2.py

```diff
@@ -219,5 +219,25 @@ def test_dataloader2_shuffle(self) -> None:
         pass
 
 
+class DataLoader2IntegrationTest(TestCase):
+
+    @staticmethod
+    def _get_multi_reading_service():
+        return MultiProcessingReadingService(num_workers=2)
+
+    def test_lazy_load(self):
+        source_dp = IterableWrapper([(i, i) for i in range(10)])
+        map_dp = source_dp.to_map_datapipe()
+        dl: DataLoader2 = DataLoader2(datapipe=map_dp, reading_service=self._get_multi_reading_service())
+        # Lazy loading
+        self.assertTrue(dl.datapipe._map is None)
+        
+        for _ in dl:
+            pass
+
+        # Lazy loading in multprocessing
+        self.assertTrue(dl.datapipe.__dict__['iterable'].dataset.__dict__['_datapipe']._map is None)
```

Comment:
Question: What is this line intending to do? Also I think it will not work when `MultiProcessingReadingService` gets replaced with the `Prototype` one.

Reply:
This line intends to get the `dl.datapipe._map` after multiprocessing.  But `dl.datapipe` changed after the `iter` function was called, so I can't use `dl.datapipe._map` to get the map attribute directly.

Yes, It can't work in `PrototyeMultiprocessingReadingService`. If it's needed, I will try to fix it.

Reply:
I think I know what you mean. 
In main process, the datapipe._map attribute should be None as you mentioned before. 
But I can't run `test_dataloader2.py`, it will get an error. Is it OK?

Besides, when I set `reading_service` to `PrototypeMultiprocessingReadingService`, It will get an exception saying Unrecognized type of request received. I think that's because some function doesn't realize in `Prototype` one, but I'm not sure.

Reply:
> But I can't run `test_dataloader2.py`, it will get an error. Is it OK?

Emm. It's not Okay since the Error apparently comes from the test. And, I am sorry that I forget to consider that `ReadingService` will adapt the DataPipe. So, here is my suggestion to the test.
```py
            # Lazy loading
            self.assertTrue(dl.datapipe._map is None)
            it = iter(dl)
            # Lazy loading in multiprocessing
            self.assertTrue(map_dp._map is None)
```

We can use `MultiprocessingReadingService` for now.

Reply:
OK, I will try it now

Reply:
I find it needs to loading data after `iter` because `iter` is just an init function and until loading data the multiprocessing will work( I guess).

If remove the loading data operation `self.assertEqual(list(it), list(range(10)))`, the test function will not work properly since I restore the deleted line in `converter.py` it still could pass.